src/v/raft/types.h

```diff
@@ -324,8 +325,12 @@ struct replicate_stages {
     // requested consistency level
     ss::future<result<replicate_result>> replicate_finished;
 };
+// leader_ack like replicate result calculated along the quorum_ack code path
+struct leader_ack_result {
+    available_promise<result<replicate_result>> offset;
+};
 
-enum class consistency_level { quorum_ack, leader_ack, no_ack };
+enum class consistency_level { quorum_ack, leader_ack, no_ack, tx_ack };
```

Comment:
i would make this a separate commit. also name `tx_ack` doesn't really say anything. Maybe we can think about something more descriptive ?